test/test_ops.py

```diff
@@ -52,12 +50,13 @@ def test_unsupported_dtypes(self, device, dtype, op):
     @onlyOnCPUAndCUDA
     @ops(op_db, dtypes=OpDTypes.supported)
     def test_supported_dtypes(self, device, dtype, op):
-        for sample in op.sample_inputs(device, dtype):
-            op(sample.input, *sample.args, **sample.kwargs)
-            # NOTE: only check the first tensor in the iterable of tensors
-            sample_input = sample.input[0] if is_iterable_of_tensors(sample.input) else sample.input
-            self.assertTrue(sample_input.dtype == dtype)
-            self.assertTrue(sample_input.device.type == self.device_type)
+        for ctx in op.sample_inputs(device, dtype):
+            with ctx as sample:
```

Comment:
An extra indentation is a pain, is there some pattern we could use to remove it?

Reply:
Maybe something from contextlib https://docs.python.org/3/library/contextlib.html? Or setting the global exception handler? Or maybe there's a class-level exception handler that works with unittest and pytest?